### PR TITLE
Dedicated module for aks run helm commands

### PR DESCRIPTION
**Pull request by dciborow**
## Description

If you haven't already, read the full [contribution guide](../CONTRIBUTING.md). The guide may have changed since the last time you read it, so please double-check. Once you are done and ready to submit your PR, edit the PR description and run through the relevant checklist below.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [x] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [x] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](../CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.

### CODE REVIEW

**Comment by dciborow** [comment]
Closes #148

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      contributor
+      rbacWriter
+    ]
+    helmApps: [{helmApp: 'azure-marketplace/wordpress', helmAppName: 'my-wordpress'}]
+  }
+}
+```
+
+```bicep
+module runCmd 'br/public:deployment-scripts/aks-run-command:1.0.1' = {
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:87]:
```suggestion
module runCmd 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
```

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    ]
+    helmApps: [{helmApp: 'azure-marketplace/wordpress', helmAppName: 'my-wordpress'}]
+  }
+}
+```
+
+### Running Helm Commands
+
+```bicep
+module helmInstallIngressController 'br/public:deployment-scripts/aks-run-command:1.0.1' = {
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:104]:
```suggestion
module helmInstallIngressController 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
```

File: modules/deployment-scripts/aks-run-helm/README.md (hunk 1 above)
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:87]:
```suggestion
module runCmd 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
```

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+module runCmd 'br/public:deployment-scripts/aks-run-command:1.0.1' = {
+  name: 'kubectlRunNginx'
+  params: {
+    aksName: aksName
+    location: location
+    rbacRolesNeeded:[
+      contributor
+      rbacWriter
+    ]
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:80]:
```suggestion
```

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+```bicep
+module runCmd 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
+  name: 'kubectlRunNginx'
+  params: {
+    aksName: aksName
+    location: location
+    rbacRolesNeeded:[
+      contributor
+      rbacWriter
+    ]
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:95]:
```suggestion
```

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+```bicep
+module helmInstallIngressController 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
+  name: 'helmInstallIngressController'
+  params: {
+    aksName: aksName
+    location: location
+    rbacRolesNeeded:[
+      contributor
+      rbacWriter
+    ]
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:112]:
```suggestion
```

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+### Sending multiple sequenced commands
+
+The module will sequence each command to run after the previous completes. There is a time impact for the DeploymentScript resource to create, therefore use new command items with consideration.
+
+```bicep
+var contributor='b24988ac-6180-42a0-ab88-20f7382dd24c'
+var rbacWriter='a7ffa36f-339b-4b5c-8bdf-e2c188b2c0eb'
+
+module runCmd 'br/public:deployment-scripts/aks-run-command:1.0.1' = {
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:72]:
```suggestion
module runCmd 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
```

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+  }
+}
+```
+
+### Using an existing managed identity
+
+When working with an existing managed identity that has the correct RBAC, you can opt out of new RBAC assignments and set the initial delay to zero.
+
+```bicep
+module runCmd 'br/public:deployment-scripts/aks-run-command:1.0.1' = {
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:50]:
```suggestion
module runCmd 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
```

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+| Name | Type | Description |
+| :--- | :--: | :---------- |
+
+## Examples
+
+### Running a simple command
+
+```bicep
+module runCmd 'br/public:deployment-scripts/aks-run-command:1.0.1' = {
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/README.md:35]:
```suggestion
module runCmd 'br/public:deployment-scripts/aks-run-helm:1.0.1' = {
```

**Review by alex-frankel — APPROVED** [review]
Great that we are re-using the more generic aks-run-command module!

File: modules/deployment-scripts/aks-run-helm/test/prereq.test.bicep [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+//   ]
+//   properties: {
+//     forceUpdateTag: forceUpdateTag
+//     azCliVersion: '2.35.0'
+//     timeout: 'PT30M'
+//     retentionInterval: 'P1D'
+//     scriptContent: 'sleep 120s'
+//     cleanupPreference: 'OnSuccess'
+//   }
+// }
```
Comment (shenglol) [inline modules/deployment-scripts/aks-run-helm/test/prereq.test.bicep:47]: Could you clean up the commented-out code?

File: modules/deployment-scripts/aks-run-helm/README.md [hunk 10]
```diff
@@ -0,0 +1,110 @@
+# AKS Run Helm Script
+
+An Azure CLI Deployment Script that allows you to run a Helm command at a Kubernetes cluster.
+
+## Description
+
+{{ Add detailed description for the module. }}
```
Comment (shenglol) [inline modules/deployment-scripts/aks-run-helm/README.md:7]: Please could you add detailed description for the module?

**Review by shenglol — CHANGES_REQUESTED** [review]
See https://github.com/Azure/bicep-registry-modules/pull/147/files#r948298354 and https://github.com/Azure/bicep-registry-modules/pull/147/files#r947347460.

File: modules/deployment-scripts/aks-run-helm/main.bicep [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+@allowed([
+  'OnSuccess'
+  'OnExpiration'
+  'Always'
+])
+@description('When the script resource is cleaned up')
+param cleanupPreference string = 'OnSuccess'
+
+module helmAppInstalls 'br/public:deployment-scripts/aks-run-command:1.0.1' = [for (app, i) in helmApps: {
+  name: 'helmInstallIngressController-${i}'
```
Comment (Gordonby) [inline modules/deployment-scripts/aks-run-helm/main.bicep:40]: Might be worth genericising the name

**Comment by dciborow** [comment]
@shenglol sorry for the delay, was out of vacation and took a little while to catch back up. Will get to this tomorrow if not tonight.

File: modules/deployment-scripts/aks-run-helm/README.md (hunk 10 above)
Comment (dciborow) [reply modules/deployment-scripts/aks-run-helm/README.md]: K, added some content and provided link for more details.

File: modules/deployment-scripts/aks-run-helm/main.bicep (hunk 11 above)
Comment (dciborow) [reply modules/deployment-scripts/aks-run-helm/main.bicep:40]:
```suggestion
  name: 'helmInstall-${cluster}-${appName}-${i}'
```

**Comment by Gordonby** [comment]
@dciborow 

You'll need to string replace the `/` with a dash or empty string etc.

image

File: modules/deployment-scripts/aks-run-helm/test/prereq.test.bicep [hunk 12]
```diff
@@ -0,0 +1,29 @@
+param aksName string =  'crtest${uniqueString(newGuid())}'
+//param forceUpdateTag  string = utcNow()
```
Comment (dciborow) [inline modules/deployment-scripts/aks-run-helm/test/prereq.test.bicep:2]:
```suggestion
```

**Comment by dciborow** [comment]
> @dciborow
> 
> In the deployment name, you'll need to string replace the `/` with a dash or empty string etc. You might also find that there's risk of exceeding the maximum deployment name length.
> 
> image

going with the app name directly now. It will not have any "/".

**Comment by dciborow** [comment]
@shenglol , any more comments on this PR?

**Review by shenglol — APPROVED** [review]
:shipit:

**Comment by shenglol** [comment]
> @shenglol , any more comments on this PR?

Nope. I was out of office on last Friday. Just merged the PR 🙂.

**Pull request merged**